api/v1alpha1/server_types.go

```diff
@@ -192,6 +210,24 @@ type NetworkInterface struct {
 	MACAddress string `json:"macAddress"`
 }
 
+// Storage defines the details of one storage device
+type Storage struct {
+	// Name is the name of the storage interface.
+	Name string `json:"name,omitempty"`
+	// Rotational specifies whether the storage device is rotational.
+	Rotational bool `json:"rotational,omitempty"`
+	// Type specifies the type of the storage device.
+	Type string `json:"type,omitempty"`
+	// SizeBytes specifies the size of the storage device in bytes.
+	Capacity resource.Quantity `json:"capacity,omitempty"`
```

Comment:
Can we use a pointer field here? That way we don't need to deref later when assigning the value.